src/mutation.cpp

```diff
@@ -318,6 +318,104 @@ bool Character::has_active_mutation( const trait_id &b ) const
     return iter != my_mutations.end() && iter->second.powered;
 }
 
+bool Character::is_category_allowed( const std::vector<std::string> &category ) const
+{
+    bool allowed = false;
+    bool restricted = false;
+    for( const trait_id &mut : get_mutations() ) {
+        for( const std::string &Ch_cat : mut.obj().allowed_category ) {
+            restricted = true;
+            for( const std::string &Mu_cat : category ) {
+                if( Ch_cat == Mu_cat ) {
+                    allowed = true;
+                }
+            }
+        }
+    }
+    if( !restricted ) {
+        allowed = true;
+    }
+    return allowed;
+
+}
+
+bool Character::is_category_allowed( const std::string &category ) const
+{
+    bool allowed = false;
+    bool restricted = false;
+    for( const trait_id &mut : get_mutations() ) {
+        for( const std::string &Ch_cat : mut.obj().allowed_category ) {
+            restricted = true;
+            if( Ch_cat == category ) {
+                allowed = true;
+            }
+        }
+    }
+    if( !restricted ) {
+        allowed = true;
+    }
+    return allowed;
+}
+
+bool Character::is_weak_to_water() const
+{
+    for( const auto &mut : my_mutations ) {
+        if( mut.first.obj().weakness_to_water > 0 ) {
+            return true;
+        }
+    }
+    return false;
+}
+
+bool Character::can_use_heal_item( itype_id heal_id ) const
```

Comment:
Considering that you always call this with `some_item.typeId()`, why not forward the item itself? That way you don't need to create a dummy item within the function, but use the actual item directly.